src/FsCheck/ArbMap.fs

```diff
@@ -1,4 +1,4 @@
-﻿namespace FsCheck
+﻿namespace rec FsCheck
```

Comment:
Can you explain why the `rec` is necessary?

Reply:
I opted for rec over TypeMapInit taking an IArbMap and assuming a downcast to ArbMap. ArbMap needs the union for the constructor, but the union needs ArbMap for the case type. I think we could reduce the scope of the recursive definition

Reply:
Reduced the recursion scope to just the two type definitions

Reply:
Reduced the recursion scope to just the two type definitions